django/db/models/functions.py

```diff
@@ -83,6 +83,28 @@ def _paired(self, expressions):
         return ConcatPair(expressions[0], self._paired(expressions[1:]))
 
 
+class Greatest(Func):
+    """
+    Chooses the maximum non-null expression and returns it.
+    """
+    function = 'GREATEST'
+
+    def __init__(self, *expressions, **extra):
+        if len(expressions) < 1:
+            raise ValueError('Greatest must take at least one expression')
+        super(Greatest, self).__init__(*expressions, **extra)
+
+    def as_sqlite(self, compiler, connection):
+        self.coalesce()
+        return super(Greatest, self).as_sql(compiler, connection, function='MAX')
+
+    def coalesce(self):
+        # any null results in null for the whole expression, wrap with coalesce
+        expressions = [
+            Coalesce(expression, Value('')) for expression in self.get_source_expressions()]
+        self.set_source_expressions(expressions)
```

Comment:
I think I've changed my mind. It's probably better to just let each backend use their own implementation. We'll need to create a backend feature though, so that different backends can test different results.

We can put a note in the documentation that lets users know that they can provide their own coalescing on backends that result in null if any argument is null.

```
Greatest(Coalesce('nullable_field', Value('default'), F('not_nullable_field'))
```

I think this simplifies the code and gives users exactly what they asked for. The drawback is obviously that their Greatest/Least code isn't fully portable - but that's sort of why we have backend features anyway.

/cc @akaariai and @mjtamlyn what do you think?